### PR TITLE
itk: add 5.1.2 and bump dependencies

### PR DESCRIPTION
**Pull request by Nekto89**
Use gdcm/3.0.9 from cci
Bump cmake_minimum_required
Add compiler check
Re-enable apple-clang - ITK 5.1.2 should support Xcode
Remove obsolete Module_ITKV3Compatibility which isn't present in ITK 5
Enable TBB

Specify library name and version:  **itk/5.1.2**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Nekto89** [comment]
Hm. It was green in draft, but now there are failures

**Review by jgsogo — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/itk/all/conanfile.py [hunk 1]
```diff
@@ -48,34 +48,55 @@ def config_options(self):
...
 
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "14",
+            "gcc": "4.8.1",
```
Comment (prince-chrismc) [inline recipes/itk/all/conanfile.py:76]: The version compare is pessimistic

```suggestion
            "gcc": "4.8",
```

**Review by prince-chrismc — APPROVED** [review]

File: recipes/itk/all/conanfile.py (hunk 1 above)
Comment (Nekto89) [reply recipes/itk/all/conanfile.py:76]: I've taken it from here: https://gcc.gnu.org/projects/cxx-status.html
"GCC 4.8.1 was the first feature-complete implementation of the 2011 C++ standard, previously known as C++0x."

Comment (prince-chrismc) [reply recipes/itk/all/conanfile.py:76]: Yes but the Conan tool will block the default settings 4.8 ... the compilers version should match the default setting (so it works for the most consumers) but it's just a note and difficult to enforce... we also have our "lazy semver" which is used by some recipes where this is important

**Pull request merged**